src/app/DefaultAttributePersistenceProvider.cpp

```diff
@@ -65,7 +65,7 @@ CHIP_ERROR DefaultAttributePersistenceProvider::ReadValue(const ConcreteAttribut
     else
     {
         // Ensure we got the expected number of bytes for all other types.
-        VerifyOrReturnError(size == aMetadata->size, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnError(size == aSize, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
This should be the INVALID_ARGUMENT error

Reply:
Should this be the same for lines 56 and 63? I'm not sure where to find the spec documentation for this.

Reply:
Look at ERROR_CODES.md in the docs directory

Reply:
`ERROS_CODES.md` is just a list of all the codes. I was hoping to find a definition of this flow that explicitly states which error should be used here.

Given that all the code is in and all tests have passed, I am going to resolve this conversation so we can merge. If you think that any of the errors should be changed, we can log an issue separately.